src/NServiceBus.Persistence.AzureTable/SagaPersisters/SecondaryIndices/SecondaryIndex.cs

```diff
@@ -85,22 +80,17 @@ public void InvalidateCache(PartitionRowKeyTuple secondaryIndexKey)
         /// <summary>
         /// Invalidates the secondary index cache if any exists for the specified property value.
         /// </summary>
-        public void InvalidateCacheIfAny(string propertyName, object propertyValue, Type sagaType)
+        public void InvalidateCacheIfAny<TSagaData>(SagaCorrelationProperty sagaCorrelationProperty)
+            where TSagaData : IContainSagaData
         {
-            var key = TryBuildKey(propertyName, propertyValue, sagaType);
-            if (key != null)
-            {
-                cache.Remove(key.Value);
-            }
+            var key = SecondaryIndexKeyBuilder.BuildTableKey<TSagaData>(sagaCorrelationProperty);
+            cache.Remove(key);
         }
 
-        static PartitionRowKeyTuple? TryBuildKey(string propertyName, object propertyValue, Type sagaType)
+        static PartitionRowKeyTuple? TryBuildKey<TSagaData>(SagaCorrelationProperty correlationProperty)
+            where TSagaData : IContainSagaData
         {
-            if (string.IsNullOrEmpty(propertyName) || propertyValue == null)
```

Comment:
I removed the null check here because the SagaCorrelationProperty has checks in the constructor already

Reply:
It's situations like this where I appreciate nullable reference types - no guesses.